internal/impl/confluent/sr/client.go

```diff
@@ -275,68 +247,3 @@ func (c *Client) walkReferencesTracked(ctx context.Context, seen map[string]int,
 	}
 	return nil
 }
-
-func (c *Client) doRequest(ctx context.Context, verb, reqPath string, body []byte) (resCode int, resBody []byte, err error) {
-	reqURL := *c.SchemaRegistryBaseURL
-	if reqURL.Path, err = url.JoinPath(reqURL.Path, reqPath); err != nil {
-		return
-	}
-
-	reqURLString := reqURL.String()
-	if match := escapedSepRegexp.MatchString(reqPath); match {
-		// Supporting '%2f' in the request url bypassing
-		// Workaround for Golang issue https://github.com/golang/go/issues/3659
-		if reqURLString, err = url.PathUnescape(reqURLString); err != nil {
-			return
-		}
-	}
-
-	var bodyReader io.Reader
-	if len(body) > 0 {
-		bodyReader = bytes.NewReader(body)
-	} else {
-		bodyReader = http.NoBody
-	}
-	var req *http.Request
-	if req, err = http.NewRequestWithContext(ctx, verb, reqURLString, bodyReader); err != nil {
-		return
-	}
-	headerKey := "Accept"
-	if verb == http.MethodPost {
-		headerKey = "Content-Type"
-	}
-	req.Header.Add(headerKey, "application/vnd.schemaregistry.v1+json")
-	if err = c.requestSigner(c.mgr.FS(), req); err != nil {
-		return
-	}
-
-	for i := 0; i < 3; i++ {
```

Comment:
Note that the `github.com/twmb/franz-go/pkg/sr` client doesn't do 3 retries under the hood. I can try to shoehorn this behaviour into the custom `roundTripper` if you think it's important to preserve it.

Reply:
Nah I think this is fine, the concept of attempts is abstract and users don't have any existing documentation or control so I think it's okay to change the behaviour slightly here.